src/python/WMCore/Storage/SiteLocalConfig.py

```diff
@@ -264,30 +278,91 @@ def processLocalStageOut():
     """
 
     while True:
-        report, node = (yield)
+        report, node, useTFC = (yield)
         localReport = {}
-        for subnode in node.children:
-            if subnode.name in ['phedex-node', 'command', 'option']:
-                localReport[subnode.name] = subnode.attrs.get('value', None)
-            elif subnode.name == 'catalog':
-                localReport[subnode.name] = subnode.attrs.get('url', None)
+        if useTFC:
+            for subnode in node.children:
+                if subnode.name in ['phedex-node', 'command', 'option']:
+                    localReport[subnode.name] = subnode.attrs.get('value', None)
+                elif subnode.name == 'catalog':
+                    localReport[subnode.name] = subnode.attrs.get('url', None)
+        else:
+            for subnode in node.children:
+                #now construct an url as used in trivial file catalog
+                subSiteName = report['subSiteName'] if 'subSiteName' in report.keys() else None
+                aStorageSite = subnode.attrs.get('site', None)
```

Comment:
Yes. I want to elaborate this a bit here. The storage site can be the site where jobs are executed or can be another site. If it is the first case the ```site``` attribute in ```<stage-out>``` is not identified. Otherwise we have defined ```site``` attribute. You can see in this example:
https://gitlab.cern.ch/SITECONF/T1_DE_KIT/-/blob/master/JobConfig/site-local-config.xml
```<method volume="KIT_dCache" protocol="WebDAV"/>```
```<method volume="KIT_dCache" protocol="SRMv2" command="gfal2"/>```
```<method site="T2_DE_DESY" volume="DESY_dCache" protocol="WebDAV"/> ```
The last choice has storage site indicated explicitly as T2_DE_DESY. Other choices the storage site is T1_DE_KIT defined in ```<site name="T1_DE_KIT">```